src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelector.java

```diff
@@ -100,11 +100,17 @@ private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Obje
                 // not a varargs call
                 return parameters;
             } else {
-                Object array = DefaultTypeTransformation.castToVargsArray(parameters, 0, parameterTypes[fixedLen]);
-                Object[] parameters2 = new Object[fixedLen + 1];
-                System.arraycopy(parameters, 0, parameters2, 0, fixedLen);
-                parameters2[fixedLen] = array;
-                return parameters2;
+                try { //if casting  is unsuccessful - just return as is
```

Comment:
Ok...so what would be ideal here is if we could determine the base type shared by all the parameters, i.e., in the example in [JENKINS-47159](https://issues.jenkins-ci.org/browse/JENKINS-47159), that'd be `ParameterValue`. Let me see what I can do there.